python/sglang/lang/interpreter.py

```diff
@@ -652,7 +653,19 @@ def _init_var_event(self, expr):
                 self._init_var_event(e)
 
     def _resolve_sampling_params(self, sampling_params):
-        clone = None
+        """
+        Construct sampling param based on default + override values
+
+        The default values of sampling are populated in `default_sampling_para` via sgl.function.run(...sampling_args)
+        , and `sampling_params` contains the override values from sgl.gen().
+
+        Here we use default_sampling_para as the base and override the values if they exist in `sampling_params`.
+        It also extends the stop tokens based on the chat template.
+        """
+
+        # deepcopy is required because the dict has lists inside
+        clone = copy.deepcopy(self.default_sampling_para)
```

Comment:
This copy is not always needed. Can we keep the original code flow so we can copy it only when needed?

Reply:
if not copied, `_handle_dtype_to_regex` might mutate the original `default_sampling_para`?